cg/services/bcl_convert_metrics_service/parser.py

```diff
@@ -175,3 +175,62 @@ def has_undetermined_reads_in_lane(self, lane: int) -> bool:
             metrics=self.quality_metrics, sample_internal_id=UNDETERMINED, lane=lane
         )
         return bool(metrics)
+
+    @classmethod
+    def calculate_total_reads_for_metrics(cls, read_pair_count: int) -> int:
+        """Scale to read pair number up to single reads."""
+        return read_pair_count * SCALE_TO_READ_PAIRS
+
+    def get_aggregate_total_reads_for_metrics(self):
+        """Return the aggregate reads for the whole demux metrics excluding indexchecks."""
+        aggregate_read_pairs: int = 0
+        for demux_metric in self.demux_metrics:
+            if demux_metric.project != "indexcheck":
+                aggregate_read_pairs += demux_metric.read_pair_count
+        return self.calculate_total_reads_for_metrics(read_pair_count=aggregate_read_pairs)
+
+    def get_aggregate_undetermined_reads_for_metrics(self) -> int:
+        """Calculate the total undermined reads to the demux metrics excluding indexchecks."""
+        aggregate_undetermined_read_pairs: int = 0
+        for demux_metric in self.demux_metrics:
+            if demux_metric.sample_internal_id == UNDETERMINED:
+                aggregate_undetermined_read_pairs += demux_metric.read_pair_count
+        return self.calculate_total_reads_for_metrics(
+            read_pair_count=aggregate_undetermined_read_pairs
+        )
+
+    def get_aggregate_percent_q30_for_metrics(self) -> float:
+        """Calculate the aggregate percent Q30 for the demux metrics excluding indexchecks."""
+        aggregate_q30_bases: int = 0
+        non_index_count: int = 0
```

Comment:
so these files also contain "indexchecks" under the "Sample_Project" column. I figured we do not want these.

Reply:
nvm we don't use the Sample_project column any longer.. we have so need to change this

Reply:
also it seems we do not report on index checks any longer. So just summing is good enough